src/ntfs/mod.rs

```diff
@@ -123,4 +156,303 @@ mod tests {
+
+        let tmp_dir = TempDir::new("navitia_model_tests").expect("create temp dir");
+
+        {
+            let path = tmp_dir.as_ref();
+            write::write_collection_with_id(path, "physical_modes.txt", &expected);
+            let col_with_id = read::make_collection_with_id(path, "physical_modes.txt");
+            assert_eq!(col_with_id, expected);
+        }
+        tmp_dir.close().expect("delete temp dir");
+    }
+
+    #[test]
+    fn routes_serialization_deserialization() {
+        let expected_objects = vec![
+            Route {
+                id: "IF:002002002:BDE".to_string(),
+                name: "Hôtels - Hôtels".to_string(),
+                direction_type: Some("foward".to_string()),
+                codes: CodesT::default(),
+                comment_links: CommentLinksT::default(),
+                line_id: "OIF:002002002:BDEOIF829".to_string(),
+                geometry_id: Some("Geometry:Line:Relation:6883353".to_string()),
+                destination_id: Some("OIF,OIF:SA:4:126".to_string()),
+            },
+            Route {
+                id: "OIF:002002002:CEN".to_string(),
+                name: "Hôtels - Hôtels".to_string(),
+                direction_type: None,
+                codes: CodesT::default(),
+                comment_links: CommentLinksT::default(),
+                line_id: "OIF:002002002:BDEOIF829".to_string(),
+                geometry_id: None,
+                destination_id: None,
+            },
+        ];
+
+        let expected = CollectionWithId::new(expected_objects);
+
+        let tmp_dir = TempDir::new("navitia_model_tests").expect("create temp dir");
+
+        {
+            let path = tmp_dir.as_ref();
+            write::write_collection_with_id(path, "routes.txt", &expected);
+            let col_with_id = read::make_collection_with_id(path, "routes.txt");
+            assert_eq!(col_with_id, expected);
+        }
+        tmp_dir.close().expect("delete temp dir");
+    }
+
+    #[test]
+    fn vehicle_journeys_and_stop_times_serialization_deserialization() {
+        let vehicle_journeys = CollectionWithId::new(vec![
+            VehicleJourney {
+                id: "OIF:87604986-1_11595-1".to_string(),
+                codes: CodesT::default(),
+                comment_links: CommentLinksT::default(),
+                route_id: "OIF:078078001:1".to_string(),
+                physical_mode_id: "Bus".to_string(),
+                dataset_id: "OIF:0".to_string(),
+                service_id: "2".to_string(),
+                headsign: Some("2005".to_string()),
+                block_id: Some("PLOI".to_string()),
+                company_id: "OIF:743".to_string(),
+                trip_property_id: Some("0".to_string()),
+                geometry_id: Some("Geometry:Line:Relation:6883353".to_string()),
+                stop_times: vec![
+                    StopTime {
+                        stop_point_idx: Idx::new(0),
```

Comment:
You can create `stop_points` before and call `stop_points.get_idx("OIF:SP:36:2085")`